source/pages/Tokens/CustomToken.tsx

```diff
@@ -104,85 +45,41 @@ export const CustomToken = () => {
-    setActiveNetwork(activeNetwork);
 
-    const contractResponse = (await contractChecker(
-      contractAddress,
-      activeNetwork.url
-    )) as ISupportsInterfaceProps;
+    try {
+      const provider = new ethers.providers.JsonRpcProvider(activeNetwork.url);
 
-    if (String(contractResponse).includes('Invalid contract address')) {
-      setErcError({
-        errorType: 'Invalid',
-        message:
-          'Invalid contract address. Verify the current contract address or the current network!',
-      });
+      const addTokenMethodResponse =
+        await controller.wallet.assets.evm.addCustomTokenByType(
+          activeAccount.address,
+          contractAddress,
+          symbol,
+          decimals,
+          provider
+        );
 
-      setIsLoading(false);
-
-      return;
-    }
-
-    switch (contractResponse.type) {
-      case 'ERC-721':
-        try {
-          const { defaultFetchValue, balanceToNumber } = await handleERC721NFTs(
-            contractAddress
-          );
-
-          if (
-            typeof balanceToNumber !== 'number' ||
-            Number.isNaN(balanceToNumber) ||
-            Boolean(String(defaultFetchValue).includes('Error'))
-          ) {
-            await handleERC20Tokens(contractAddress, decimals);
-
-            return;
-          }
-
-          const treatedSymbol = symbol.replaceAll(/\s/g, '').toUpperCase();
-
-          await controller.wallet.account.eth.saveTokenInfo({
-            tokenSymbol: treatedSymbol,
-            contractAddress,
-            decimals,
-            isNft: true,
-            balance: balanceToNumber,
-          });
-          setIsLoading(false);
-          setAdded(true);
-
-          return;
-        } catch (_erc721Error) {
-          setIsLoading(false);
-          setErcError({
-            errorType: 'Undefined',
-            message: '',
-          });
-        }
-        break;
-      case 'ERC-1155':
+      if (addTokenMethodResponse.error) {
         setIsLoading(false);
         setErcError({
-          errorType: 'ERC-1155',
-          message: contractResponse.message,
+          errorType: addTokenMethodResponse.errorType,
+          message: addTokenMethodResponse.message,
         });
-        break;
-      default:
-        // Default will be for cases when contract type will come as Undefined. This type is for ERC-20 cases or contracts that type
-        // has not been founded
-        try {
-          await handleERC20Tokens(contractAddress, decimals);
-          setIsLoading(false);
-          return;
-        } catch (_ercUndefinedError) {
-          setIsLoading(false);
-          setErcError({
-            errorType: 'Undefined',
-            message: '',
-          });
-        }
-        break;
+
+        return;
+      }
+
+      await controller.wallet.account.eth.saveTokenInfo(
+        addTokenMethodResponse.tokenToAdd
+      );
+
+      setIsLoading(false);
+      setAdded(true);
+    } catch (error) {
```

Comment:
instead of calling `setIsLoading(false);` you can add it on the `finally` after the `catch` statement